helpdesk-maker-checker-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/makerchecker/MakerCheckerService.java

```diff
@@ -63,60 +63,45 @@ public boolean allChecksPass(SymMessage symMessage) {
   }
 
   /**
-   * Accept a maker checker message.
+   * Approve a maker checker message.
    * Find the message.
    * Send the message to client stream.
    * @param makerCheckerMessage the maker checker message
    */
-  public Set<SymMessage> getAcceptMessages(MakerCheckerMessage makerCheckerMessage) {
-    SymStream stream = new SymStream();
-    stream.setStreamId(makerCheckerMessage.getStreamId());
+  public Set<SymMessage> getApprovedMakercheckerMessage(MakerCheckerMessage makerCheckerMessage) {
     try {
-      List<SymMessage> symMessageList = symphonyClient.getMessagesClient()
-          .getMessagesFromStream(stream, makerCheckerMessage.getTimeStamp() - 1, 0, 10);
+      SymStream stream = new SymStream();
+      stream.setStreamId(makerCheckerMessage.getStreamId());
 
-      SymMessage match = null;
-      for(SymMessage symMessage : symMessageList) {
-        if(symMessage.getId().equals(makerCheckerMessage.getMessageId())) {
-          match = symMessage;
-        }
-      }
-
-      if(match == null) {
-        throw new BadRequestException(String.format(MESSAGE_NOT_FOUND, makerCheckerMessage.getMessageId()));
-      }
+      SymMessage symMessage = getApprovedMessage(makerCheckerMessage, stream);
+      Set<SymMessage> symApprovedMessages = null;
 
-      Set<SymMessage> acceptMessages = new HashSet<>();
       for(Checker checker: checkerSet) {
         if(checker.isCheckerType(makerCheckerMessage)) {
-          Set<SymMessage> messages = checker.makeApprovedMessages(makerCheckerMessage, match);
-          for(SymMessage symMessage: messages) {
-            for(String streamId: makerCheckerMessage.getProxyToStreamIds()) {
-              stream = new SymStream();
-              stream.setStreamId(streamId);
-
-              SymMessage copy = new SymMessage();
-              copy.setStreamId(streamId);
-              copy.setStream(stream);
-              copy.setMessage(symMessage.getMessage());
-              copy.setEntityData(symMessage.getEntityData());
-              copy.setAttachments(symMessage.getAttachments());
-              copy.setTimestamp(symMessage.getTimestamp());
-              copy.setFromUserId(symMessage.getFromUserId());
-
-              acceptMessages.add(copy);
-            }
-          }
+          symApprovedMessages = checker.makeApprovedMessages(makerCheckerMessage, symMessage);
```

Comment:
You should add all objects to this collection instead of overriding it.

Reply:
Done.